### PR TITLE
feat: parse HTML element and attribute positions

### PR DESCRIPTION
**Pull request by alehechka**
# Overview
This PR adds positions to the templ parser results for HTML elements and attributes relating to conversation in  https://github.com/a-h/templ/issues/498, specifically the recommendation made in https://github.com/a-h/templ/issues/498#issuecomment-1952638089.

This handles only calculating expression positions. I plan to open another PR when ready to use the calculated positions to enable LSP intellisense with a few hard-coded results. After that, I can work towards the solution I recommended in https://github.com/a-h/templ/issues/498#issuecomment-1952566968, to import an HTML LSP's data into templ's LSP.

### CODE REVIEW

File: parser/v2/types.go [hunk 1]
```diff
@@ -405,6 +405,17 @@ type WhitespaceTrailer interface {
...
+type HTMLExpression interface {
+	HTMLExpression() Expression
+}
+
+var (
+	_ HTMLExpression = Element{}
+	_ HTMLExpression = BoolConstantAttribute{}
+	_ HTMLExpression = BoolExpressionAttribute{}
+	_ HTMLExpression = ConstantAttribute{}
+)
```
Comment (alehechka) [inline parser/v2/types.go:417]: I added this with plans to eventually use the interface to determine when the HTML LSP can be enabled for intellisense.

**Comment by alehechka** [comment]
@a-h, I finished getting all the unit tests updated to reflect the position calculations so this should be ready for code review now.

**Comment by joerdav** [comment]
My immediate thoughts here are that maybe `Expression` isn't the right type, so far I think we only store Go in here.

I could be being nit-picky but I think it makes more sense for there to be an HTML specific expression type.

**Comment by alehechka** [comment]
@joerdav, that sounds valid. I mainly reused it since it was already a pattern for storing positions. Should be simple enough to do a find and replace with a new name. Do you have anything in mind as an alternative to `Expression`?

**Comment by alehechka** [comment]
Considering that `Expression.Value` ends up with the same value as `Name` from the parent struct in all of these cases, we could probably just use `Range` directly to store the positions of elements/attributes.

**Comment by alehechka** [comment]
@joerdav, I went ahead and replaced `Expression` with `Range` in all cases. Let me know if this approach works better with you.

**Comment by joerdav** [comment]
@alehechka the type range makes sense, happy with that. I wonder if a better name for the property could be in order? The name `Range Range` to me suggests it's the range of the whole open tag, maybe `NameRange Range` could be more specific? After that I'm happy with this PR if @a-h is.

**Review by a-h — APPROVED** [review]

**Pull request merged**